tests/framework/microvm.py

```diff
@@ -480,34 +481,36 @@ def spawn(
                 {"metadata": os.path.basename(self.metadata_file)}
             )
 
-        jailer_param_list = self.jailer.construct_param_list()
-
         if log_level != "Debug":
             # Checking the timings requires DEBUG level log messages
             self.time_api_requests = False
 
         # When the daemonize flag is on, we want to clone-exec into the
-        # jailer rather than executing it via spawning a shell. Going
-        # forward, we'll probably switch to this method for running
-        # Firecracker in general, because it represents the way it's meant
-        # to be run by customers (together with CLONE_NEWPID flag).
-        #
-        # We have to use an external tool for CLONE_NEWPID, because
-        # 1) Python doesn't provide os.clone() interface, and
-        # 2) Python's ctypes libc interface appears to be broken, causing
-        # our clone / exec to deadlock at some point.
+        # jailer rather than executing it via spawning a shell.
         if self.jailer.daemonize:
-            self.daemonize_jailer(jailer_param_list)
+            res = subprocess.Popen(
+                [str(self._jailer_binary_path)] + self.jailer.construct_param_list(),
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            stdout, stderr = res.communicate()
+            if res.returncode != 0:
+                raise RuntimeError(res.returncode, stdout, stderr)
```

Comment:
Why not simply `utils.run_cmd(blablabla)`?

Reply:
Heh, I didn't know it does exactly this. Not sure. In this case I wanted full control of the process.